drivers/gnss/gnss_u_blox_m10.c

```diff
@@ -0,0 +1,940 @@
+		return GNSS_NAVIGATION_MODE_HIGH_DYNAMICS;
+	case UBX_DYN_MODEL_WIRST:
+		return GNSS_NAVIGATION_MODE_BALANCED_DYNAMICS;
+	case UBX_DYN_MODEL_BIKE:
+		return GNSS_NAVIGATION_MODE_HIGH_DYNAMICS;
+	default:
+		return -EINVAL;
+	}
+}
+
+static int ubx_m10_set_navigation_mode(const struct device *dev, enum gnss_navigation_mode mode)
+{
+	int ret;
+	k_spinlock_key_t key;
+	struct ubx_m10_data *data = dev->data;
+	struct ubx_cfg_nav5_data payload;
+
+	key = k_spin_lock(&data->lock);
+
+	(void) ubx_cfg_nav5_data_default(&payload);
+
+	ret = ubx_m10_navigation_mode_to_ubx_dynamic_model(dev, mode);
+	if (ret < 0) {
+		goto unlock;
+	}
+
+	payload.dyn_model = ret;
+
+	ret = ubx_m10_modem_ubx_script_init(dev, &payload, UBX_CFG_NAV5_PAYLOAD_SZ, UBX_CLASS_CFG,
+					    UBX_CFG_NAV5);
+	if (ret < 0) {
+		goto unlock;
+	}
+
+	ret = ubx_m10_modem_ubx_run_script(dev, &(data->script));
+	if (ret < 0) {
+		goto unlock;
+	}
+
+	k_sleep(K_MSEC(UBX_CFG_NAV5_WAIT_MS));
+
+unlock:
+	k_spin_unlock(&data->lock, key);
+
+	return ret;
+}
+
+static int ubx_m10_get_navigation_mode(const struct device *dev, enum gnss_navigation_mode *mode)
+{
+	int ret;
+	k_spinlock_key_t key;
+	struct ubx_m10_data *data = dev->data;
+
+	key = k_spin_lock(&data->lock);
+
+	ret = ubx_m10_modem_ubx_script_init(dev, NULL, UBX_FRM_GET_PAYLOAD_SZ, UBX_CLASS_CFG,
+					    UBX_CFG_NAV5);
+	if (ret < 0) {
+		goto unlock;
+	}
+
+	ret = ubx_m10_modem_ubx_run_script(dev, &(data->script));
+	if (ret < 0) {
+		goto unlock;
+	}
+
+	struct ubx_frame_t *response = data->script.response;
+
+	enum ubx_dynamic_model dynamic_model =
+		((struct ubx_cfg_nav5_data *)response->payload_and_checksum)->dyn_model;
+	ret = ubx_m10_ubx_dynamic_model_to_navigation_mode(dev, dynamic_model);
+	if (ret < 0) {
+		goto unlock;
+	}
+
+	*mode = ret;
+
+unlock:
+	k_spin_unlock(&data->lock, key);
+
+	return ret;
+}
+
+static int ubx_m10_get_supported_systems(const struct device *dev, gnss_systems_t *systems)
+{
+	*systems = (GNSS_SYSTEM_GPS | GNSS_SYSTEM_GLONASS | GNSS_SYSTEM_GALILEO |
+		    GNSS_SYSTEM_BEIDOU | GNSS_SYSTEM_SBAS | GNSS_SYSTEM_QZSS);
+
+	return 0;
+}
+
+static int ubx_m10_ubx_gnss_id_to_gnss_system(const struct device *dev, enum ubx_gnss_id gnss_id)
+{
+	switch (gnss_id) {
+	case UBX_GNSS_ID_GPS:
+		return GNSS_SYSTEM_GPS;
+	case UBX_GNSS_ID_SBAS:
+		return GNSS_SYSTEM_SBAS;
+	case UBX_GNSS_ID_GALILEO:
+		return GNSS_SYSTEM_GALILEO;
+	case UBX_GNSS_ID_BEIDOU:
+		return GNSS_SYSTEM_BEIDOU;
+	case UBX_GNSS_ID_QZSS:
+		return GNSS_SYSTEM_QZSS;
+	case UBX_GNSS_ID_GLONAS:
+		return GNSS_SYSTEM_GLONASS;
+	default:
+		return -EINVAL;
+	};
+}
+
+static int ubx_m10_config_block_fill(const struct device *dev, gnss_systems_t gnss_system,
+	struct ubx_cfg_gnss_data *payload, uint8_t index, bool enable)
+{
+	switch (gnss_system) {
+	case GNSS_SYSTEM_GPS:
+		payload->config_blocks[index].gnss_id = UBX_GNSS_ID_GPS;
+		payload->config_blocks[index].flags = enable |
+			UBX_CFG_GNSS_FLAG_SGN_CNF_GPS_L1C_A;
```

Comment:
is this correct? boolean | value? guess it's fine it just looks odd to me, maybe add a define for the enable/disable bit and use an intermediate variable or something, just for clarity

Reply:
Hi @fabiobaltieri, i have addressed this comment in the latest push. Kindly check if the resolution is correct.